invenio_files_rest/views.py

```diff
@@ -802,6 +835,49 @@ def delete(self, bucket=None, key=None, version_id=None, upload_id=None,
             obj = self.get_object(bucket, key, version_id)
             return self.delete_object(bucket, obj, version_id)
 
+
+class TagResource(ContentNegotiatedMethodView):
+    """Tag resource."""
+
+    def __init__(self, *args, **kwargs):
+        """Instantiate content negotiated view."""
+        super(TagResource, self).__init__(*args, **kwargs)
+
+    @pass_bucket
+    @need_bucket_permission('bucket-update')
```

Comment:
@lnielsen suggested that tags need a finer permission granularity (``tag-update``)

Reply:
do you think it's better something like `objectversion-tag-update`?

Reply:
@hachreak yup

Reply:
especially w.r.t. https://github.com/inveniosoftware/invenio-files-rest/pull/167#discussion_r138885083